src/ReservoirPair.sol

```diff
@@ -512,13 +514,18 @@ abstract contract ReservoirPair is IAssetManagedPair, ReservoirERC20 {
             return (aCurrRawPrice, int112(LogCompression.toLowResLog(aCurrRawPrice)));
         }
 
+        // call to `percentDelta` is safe as long as the difference between aCurrRawPrice and aPrevClampedPrice is
+        // less than uint196 (1e59). It is extremely unlikely that one trade can change the price by 1e59
         if (aCurrRawPrice.percentDelta(aPrevClampedPrice) > maxChangeRate * aTimeElapsed) {
             // clamp the price
+            // multiplication of maxChangeRate and aTimeElapsed would not overflow as
+            // maxChangeRate <= 0.01e18 (50 bits)
+            // aTimeElapsed <= 32 bits
             if (aCurrRawPrice > aPrevClampedPrice) {
-                rClampedPrice = aPrevClampedPrice * (1e18 + (maxChangeRate * aTimeElapsed)) / 1e18;
+                rClampedPrice = aPrevClampedPrice.mulDiv(1e18 + maxChangeRate * aTimeElapsed, 1e18);
```

Comment:
max price the oracle can record is `e^130` which is around 248 bits. So phantom overflow is possible in this case, thus the fix

Reply:
Makes sense